#! /bin/sh /usr/share/dpatch/dpatch-run
## 02_video_device_2_6_26_27.dpatch by  <alban.peignier@free.fr>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Skip video_device obsolete fields for kernel >= 2.6.24

@DPATCH@

diff --git a/v4l/asihpi-v4l.c b/v4l/asihpi-v4l.c
index d11a9b7..2bb802e 100644
--- a/v4l/asihpi-v4l.c
+++ b/v4l/asihpi-v4l.c
@@ -561,11 +561,17 @@ int init_module(void)
 	      }
 	      k++;
 	    }
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 27)
 	    radio_asihpi_tuner[i][j].videodev.owner=THIS_MODULE;
+#endif
 	    sprintf(radio_asihpi_tuner[i][j].videodev.name,
 		    "AudioScience %4x[%d] Tuner %d",adapter_list[i],i+1,j+1);
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 27)
 	    radio_asihpi_tuner[i][j].videodev.type=VID_TYPE_TUNER;
+#endif
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 24)
 	    radio_asihpi_tuner[i][j].videodev.hardware=VID_HARDWARE_ASIHPI;
+#endif
 	    radio_asihpi_tuner[i][j].videodev.fops=&radio_asihpi_fops;
 
 #if LINUX_VERSION_CODE >= 0x20500
